src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorDocumentManager.ts

```diff
@@ -17,19 +18,33 @@ import { RazorLogger } from './RazorLogger';
 import { UpdateBufferRequest } from './RPC/UpdateBufferRequest';
 import { getUriPath } from './UriPaths';
 
-export class RazorDocumentManager implements IRazorDocumentManager {
+export class RazorDocumentManager implements IRazorDocumentManager, StaticFeature {
+
+    public get onChange() { return this.onChangeEmitter.event; }
+
+    public get documents() {
+        return Object.values(this.razorDocuments);
+    }
+    public fillInitializeParams?: ((params: InitializeParams) => void) | undefined;
     private readonly razorDocuments: { [hostDocumentPath: string]: IRazorDocument } = {};
     private onChangeEmitter = new vscode.EventEmitter<IRazorDocumentChangeEvent>();
 
     constructor(
         private readonly serverClient: RazorLanguageServerClient,
         private readonly logger: RazorLogger) {
     }
+    public fillClientCapabilities(capabilities: ClientCapabilities): void {
+        return;
+    }
 
-    public get onChange() { return this.onChangeEmitter.event; }
+    public async initialize(capabilities: ServerCapabilities, documentSelector: DocumentSelector | undefined): Promise<void> {
```

Comment:
Perhaps the `Feature` should be a separate class which takes an `IRazorDocumentManager`.